process/throttle/antiflood/p2pAntiflood.go

```diff
@@ -75,22 +75,22 @@ func (af *p2pAntiflood) CanProcessMessage(message p2p.MessageP2P, fromConnectedP
 	return nil
 }
 
-// CanProcessMessageOnTopic signals if a p2p message can be processed or not for a given topic
-func (af *p2pAntiflood) CanProcessMessageOnTopic(peer p2p.PeerID, topic string) error {
+// CanProcessMessagesOnTopic signals if a p2p message can be processed or not for a given topic
+func (af *p2pAntiflood) CanProcessMessagesOnTopic(peer p2p.PeerID, topic string, numMessages uint32) error {
```

Comment:
If this is not returning a bool and it has side effects (it actually increments some state) could we name it something else?

Reply:
Will think about a better name

Reply:
Thought of it, maybe ReceivedMessage might be a better match but it does not tell much as what it does. CanProcessMessage actually tells that the message should be processed or not. Agree it does change it's internal state but that is why it was constructed, no way to circumvent it. It has to decide if the message should be processed or not based on some state (that might change after this call). @sasurobert ?

Reply:
This is like a part 2 of my earlier question regarding incrementing than doing the check :)
For example, I would have seen it like:
```go
func CanProcessMessage(uint size) bool {
  if currentState + size > limit {
    return false
  }
  return true
}

if CanProcessMessage(someSize) {
  increaseCurrentState(someSize)
  processMessage()
}
```

This way if the limit is 10 messages, I already sent 8, then I have 2 requests, one with 3 and one with 2 messages, the one with 2 will be processed. In the current way, neither will be processed, because `CanProcessMessage` will increase the state to 11, so until a state reset the peer is blocked.

But it's ok like this also, I just wanted to write a follow up to better explain what I meant 👍

Reply:
Totally agree that the functionality can be split, somehow, because the function does 2 things, but how our project evolves, someone could just called CanProcessMessage without calling increaseCurrentState resulting in an ineffective mechanism that would have been hard to debug. So, at the expense of SRP, the functions does a checking and a state change.